### PR TITLE
Aliases for DB software

### LINKED ISSUE #395: Database software type value input should be more flexible
## Issue

Running the following command:

```bash
civo database create db-psql --software psql
```

Gives this output:

```bash
Please check if you are using the latest version of CLI and retry the command 
If you are still facing issues, please report it on our community slack or open a GitHub issue (https://github.com/civo/cli/issues) 
Error: The provided software name is not valid. Make sure you use correct capitalization (e.g. MySQL, PostgreSQL)
```

Aside from the [generic error which is useless](https://github.com/civo/cli/issues/391) in this scenario, there are a couple of issues here:

### 1. The values that can be provided here should be more flexible. 

They should not be case sensitive, and they should accept shorthand form of the db name. For example it should accept `postgresql` as well as `PostgreSQL` and `psql` and `PSQL`. This would make the usage more intuitive.

### 2. The `--software` flag is a little strange, wouldn't `--type` be more fitting?

This one is as big of a deal, but I feel that the conceptual model of the user is more inclined to use `--type` as a flag for this than `--software`

## Acceptance Criteria

* Accept shorthands for databases and make the names case insensitive.
* Reconsider the name of the flag --> Not as important, required to close this ticket.

### PR DESCRIPTION
**Pull request by uzaxirr**
Fixes: #395
Screenshot 2024-04-30 at 6 29 49 AM

### CODE REVIEW

File: cmd/database/database_create.go [hunk 1]
```diff
@@ -87,45 +88,51 @@ var dbCreateCmd = &cobra.Command{
...
-			software = "MySQL"
-		}
+		software = strings.ToLower(software)
+		softwareIsValid := false
+		softwareVersionIsValid := false
 
-		if softwareVersion == "" && software == "MySQL" {
-			softwareVersion = "8.0"
+		validSoftwares := map[string][]string{
+			"mysql":      {"mysql", "my", "sql"},
```
Comment (vishalanarase) [inline cmd/database/database_create.go]: I don't think we should allow "my" and "sql"

Comment (uzaxirr) [reply cmd/database/database_create.go]: https://github.com/civo/cli/pull/413/commits/b672f2650e5875f60f871107e45edb5862036c5a

File: cmd/database/database_create.go [hunk 2]
```diff
@@ -87,45 +88,51 @@ var dbCreateCmd = &cobra.Command{
...
 		if !softwareIsValid {
-			utility.Error("The provided software name is not valid. Make sure you use correct capitalization (e.g. MySQL, PostgreSQL)")
+			utility.Error("The provided software name is not valid. Make sure you use correct capitalization (e.g., MySQL, PostgreSQL)")
 			os.Exit(1)
 		}
 
 		if !softwareVersionIsValid {
-			utility.Error("The provided software version is not valid")
+			if softwareVersion == "" {
+				utility.Error(fmt.Sprintf("No version specified for %s. Please provide a version using --version flag. For example, civo database create db-psql --software psql --version 14.", canonicalSoftwareName))
```
Comment (alessandroargentieri) [inline cmd/database/database_create.go]: please remove the .

File: cmd/database/database_create.go [hunk 3]
```diff
@@ -87,45 +88,51 @@ var dbCreateCmd = &cobra.Command{
...
 						}
 					}
 				}
-
 			}
 		}
 
 		if !softwareIsValid {
-			utility.Error("The provided software name is not valid. Make sure you use correct capitalization (e.g. MySQL, PostgreSQL)")
+			utility.Error("The provided software name is not valid. Make sure you use correct capitalization (e.g., MySQL, PostgreSQL)")
```
Comment (alessandroargentieri) [inline cmd/database/database_create.go]: please remove comma

File: cmd/database/database_create.go [hunk 4]
```diff
@@ -87,45 +88,51 @@ var dbCreateCmd = &cobra.Command{
...
 						}
 					}
 				}
-
 			}
 		}
 
 		if !softwareIsValid {
-			utility.Error("The provided software name is not valid. Make sure you use correct capitalization (e.g. MySQL, PostgreSQL)")
+			utility.Error("The provided software name is not valid. Make sure you use correct capitalization (eg: MySQL, PostgreSQL)")
```
Comment (fernando-villalba) [inline cmd/database/database_create.go]: I am confused,  if the change was to allow the usage of lowercase names of the database (ideally case insensitive) why do we still have this error asking for correct capitalization?

**Review by fernando-villalba — CHANGES_REQUESTED** [review]
Change error message so it doesn't mislead the user to think capitalization is still needed.

**Review by fernando-villalba — APPROVED** [review]
LGTM!

**Pull request merged**